controllers/apps/transformer_cluster_backup_policy.go

```diff
@@ -444,67 +452,78 @@ func (r *clusterBackupPolicyTransformer) syncBackupMethods(backupPolicy *dpv1alp
 	backupPolicy.Spec.BackupMethods = backupMethods
 }
 
-func (r *clusterBackupPolicyTransformer) doEnvMapping(comp *appsv1.ClusterComponentSpec, envMapping []appsv1alpha1.EnvMappingVar) []corev1.EnvVar {
+func (r *backupPolicyBuilder) resolveBackupMethodEnv(compSpec *appsv1.ClusterComponentSpec, envs []dpv1alpha1.EnvVar) []corev1.EnvVar {
 	var env []corev1.EnvVar
-	for _, v := range envMapping {
-		for _, cm := range v.ValueFrom.ComponentDef {
-			if !slices.Contains(cm.Names, comp.ComponentDef) {
-				continue
+	for _, v := range envs {
+		if v.Value != nil {
+			env = append(env, corev1.EnvVar{Name: v.Name, Value: *v.Value})
+			continue
+		}
+		if v.ValueFrom != nil {
+			for _, versionMapping := range v.ValueFrom.VersionMapping {
+				if r.matchMappingName(versionMapping.ServiceVersions, compSpec.ServiceVersion) {
+					env = append(env, corev1.EnvVar{Name: v.Name, Value: versionMapping.MappedValue})
+					break
+				}
 			}
-			env = append(env, corev1.EnvVar{
-				Name:  v.Key,
-				Value: cm.MappingValue,
-			})
 		}
 	}
 	return env
 }
 
-func (r *clusterBackupPolicyTransformer) syncBackupPolicyTargetSpec(backupPolicy *dpv1alpha1.BackupPolicy, comp componentItem) {
-	if comp.isSharding {
-		backupPolicy.Spec.Targets = r.buildBackupTargets(backupPolicy.Spec.Targets, comp)
+func (r *backupPolicyBuilder) matchMappingName(names []string, target string) bool {
+	for _, name := range names {
+		if component.CompDefMatched(target, name) {
```

Comment:
`CompDefMatched()` is used for matching compDef name, so it's unsuitable here.

Maybe a more generic function can be extracted (e.g.: `util.PrefixOrRegexMatched()`), then `CompDefMatched()` can use this function too.